plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -85,24 +84,27 @@ public void loadConnectors(Settings settings) {
   }
 
   @Override
-  public StorageEngine getStorageEngine(String catalog) {
-    if (!storageEngineMap.containsKey(catalog)) {
-      return defaultOpenSearchStorageEngine;
-    }
-    return storageEngineMap.get(catalog);
+  public Set<Catalog> getCatalogs() {
+    return new HashSet<>(catalogMap.values());
   }
 
   @Override
-  public Set<String> getCatalogs() {
-    return Collections.unmodifiableSet(storageEngineMap.keySet());
+  public Catalog getCatalog(String catalogName) {
+    if (!catalogMap.containsKey(catalogName)) {
+      throw new IllegalArgumentException(
+          String.format("Catalog with name %s doesn't exist.", catalogName));
+    }
+    return catalogMap.get(catalogName);
   }
 
+
   @Override
   public void registerOpenSearchStorageEngine(StorageEngine storageEngine) {
```

Comment:
This may be changed to `registerCatalog` or `createCatalog`?

Reply:
will changed the name to `registerDefaultOpensearchCatalog.`

Reply:
Can we make this API method more generic? Why the default catalog has to be OpenSearch in future?

Also I see that you renamed it, however, the method still accepts `StorageEngine` rather than `Catalog`?

Reply:
`| rest /services/data/indexes datatype=all` The above one for spl for listing all indices.  Couldn't figure out any command for listing. There is `mcatalog` which works on metric type of indices.

Reply:
This method is created to handle a special case for opensearch catalog.  `For Now I want to keep the intent clear from the function name that this is a special case`.

Also there is expectation that this method runs after every loadConnectors method. 
When we redesign catalogs(Introduce Rest APIs maybe) in the next phase we can create a generic register api if required.